cmd/zc_traverser_list.go

```diff
@@ -93,7 +93,7 @@ func (l *listTraverser) Traverse(preprocessor objectMorpher, processor objectPro
 func newListTraverser(parent common.ResourceString, parentType common.Location, credential *common.CredentialInfo,
 	ctx *context.Context, recursive bool, handleSymlinks common.SymlinkHandlingType, getProperties bool, listChan chan string,
 	includeDirectoryStubs bool, incrementEnumerationCounter enumerationCounterFunc, s2sPreserveBlobTags bool,
-	logLevel pipeline.LogLevel, cpkOptions common.CpkOptions, syncHashType common.SyncHashType, preservePermissions common.PreservePermissionsOption, trailingDot common.TrailingDotOption, p pipeline.Pipeline) ResourceTraverser {
+	logLevel pipeline.LogLevel, cpkOptions common.CpkOptions, syncHashType common.SyncHashType, preservePermissions common.PreservePermissionsOption, trailingDot common.TrailingDotOption, p pipeline.Pipeline, destination *common.Location) ResourceTraverser {
```

Comment:
As soon as the track 2 migration is merged, we need to have a discussion as a team about how we want to handle passing around flags in AzCopy in the future. 20+ params is unweildly and often an options bag would at least be more maintainable, but still suffer from the issue of mentally having to calculate all the layers a flag has to get copied through before use.